src/Services/SchemaPoster.php

```diff
@@ -23,369 +23,382 @@
+    {
+        try {
+            $uri = (new Uri(
+                $this->getClient()
+                    ->getEndpoint()
+                    ->getCoreBaseUri() . 'admin/file'
+            ))->withQuery(
+                http_build_query([
+                    'action' => 'VIEW',
+                    'file' => $filename,
+                ])
+            );
+            $this->logger->debug('Upload url: ' . $uri);
+            $request = new Request('GET', $uri);
+            $response = $this->client->sendRequest($request);
+            $message = vsprintf('File: %s, Status code: %d - %s', [
+                'filename' => $filename,
+                'status_code' => $response->getStatusCode(),
+                'reason' => $response->getReasonPhrase(),
+            ]);
+            $this->logger->debug($message);
 
-  /**
-   * Set Verbosity.
-   *
-   * @param bool $isVerbose
-   *   Verbosity value.
-   */
-  public function setVerbose(bool $isVerbose): void {
-    $this->verbose = $isVerbose;
-  }
+            return $response->getBody();
+        } catch (\Throwable $e) {
+            $message = vsprintf('File: %s, Status code: %d - %s', [
+                'filename' => $filename,
+                'status_code' => $e->getCode(),
+                'reason' => $e->getMessage(),
+            ]);
+            $this->logger->error($message);
+        }
+
+        return null;
+    }
 
-  /**
-   * Get Logger Instance.
-   *
-   * @return \Psr\Log\LoggerInterface
-   *   Drupal's Logger Interface.
-   */
-  public function getLogger() {
-    return $this->logger;
-  }
 
-  /**
-   * Set Logger Instance.
-   *
-   * @param \Psr\Log\LoggerInterface $logger
-   *   Drupal's Logger Interface.
-   */
-  public function setLogger(LoggerInterface $logger): void {
-    $this->logger = $logger;
-  }
+    public function uploadOneAtATime(string $server_id)
+    {
+        $schemaFiles = $this->getSolrFiles($server_id);
+        $toReturn = [];
+        foreach ($schemaFiles as $filename => $contents) {
+            $contentType =
+                substr($filename, 0, -3) == '.xml' ? 'application/xml' : 'text/plain';
+            $response = $this->getClient()->post(
+                $this->getClient()
+                    ->getEndpoint()
+                    ->getSchemaUploadUri(),
+                [
+                    'query' => [
+                        'action' => 'UPLOAD',
+                        'name' => '_default',
+                        'filePath' => $filename,
+                        'contentType' => $contentType,
+                        'overwrite' => 'true',
+                    ],
+                    'headers' => [
+                        'Content-Type' => 'application/octet-stream',
+                    ],
+                    'body' => $contents,
+                ]
+            );
+            // Parse the response.
+            $log_function = in_array($response->getStatusCode(), [200, 201, 202, 203])
```

Comment:
Useless extra check, `guzzle` will throw an exception for anything non-2xx. Should be converted int `try/catch` block.

Reply:
See above.